src/Lykke.Service.Dash.Api.Services/DashService.cs

```diff
@@ -0,0 +1,310 @@
+            BitcoinAddress toAddress, decimal amount, bool includeFee)
+        {
+            var sendAmount = Money.FromUnit(amount, Asset.Dash.Unit);
+            var txsUnspent = await _dashInsightClient.GetTxsUnspentAsync(fromAddress.ToString());
+
+            var builder = new TransactionBuilder()
+                .Send(toAddress, sendAmount)
+                .SetChange(fromAddress)
+                .SetTransactionPolicy(new StandardTransactionPolicy
+                {
+                    CheckFee = false
+                });
+
+            if (includeFee)
+            {
+                builder.SubtractFees();
+            }
+
+            foreach (var txUnspent in txsUnspent)
+            {
+                var coin = new Coin(
+                    fromTxHash: uint256.Parse(txUnspent.Txid),
+                    fromOutputIndex: txUnspent.Vout,
+                    amount: Money.Coins(txUnspent.Amount),
+                    scriptPubKey: fromAddress.ScriptPubKey);
+
+                builder.AddCoins(coin);
+            }
+
+            var feeMoney = Money.FromUnit(_dashApiSettings.Fee, Asset.Dash.Unit);
+
+            var tx = builder
+                .SendFees(feeMoney)
+                .BuildTransaction(false);
+
+            var coins = builder.FindSpentCoins(tx);
+
+            return Serializer.ToString((tx: tx, coins: coins));
+        }
+
+        public async Task BroadcastAsync(Transaction transaction, Guid operationId)
+        {
+            TxBroadcast response;
+
+            try
+            {
+                response = await _dashInsightClient.BroadcastTxAsync(transaction.ToHex());
+
+                if (response == null)
+                {
+                    throw new ArgumentException($"{nameof(response)} can not be null");
+                }
+                if (string.IsNullOrEmpty(response.Txid))
+                {
+                    throw new ArgumentException($"{nameof(response)}{nameof(response.Txid)} can not be null or empty. Response={response}");
+                }
+            }
+            catch (Exception ex)
+            {
+                await _log.WriteErrorAsync(nameof(DashService), nameof(BroadcastAsync),
+                    $"transaction: {transaction.ToString()}, operationId: {operationId}", ex);
+
+                throw;
+            }
+
+            var block = await GetLatestBlockHeight();
+
+            await _broadcastRepository.AddAsync(operationId, response.Txid, block);
+            await _broadcastInProgressRepository.AddAsync(operationId, response.Txid);
+        }
+
+        public async Task<IBroadcast> GetBroadcastAsync(Guid operationId)
+        {
+            return await _broadcastRepository.GetAsync(operationId);
+        }
+
+        public async Task DeleteBroadcastAsync(IBroadcast broadcast)
+        {
+            await _broadcastRepository.DeleteAsync(broadcast.OperationId);
```

Comment:
If execution fail after this line, data from _broadcastInProgressRepository will never be deleted